server/src/main/java/org/elasticsearch/index/engine/Segment.java

```diff
@@ -237,6 +244,18 @@ private static void writeSegmentSort(StreamOutput out, Sort sort) throws IOExcep
                 o.writeGenericValue(field.getMissingValue());
                 o.writeBoolean(((SortedNumericSortField) field).getSelector() == SortedNumericSelector.Type.MAX);
                 o.writeBoolean(field.getReverse());
+            } else if (field.getType().equals(SortField.Type.STRING)) {
+                if (o.getVersion().before(Version.V_8_5_0)) {
+                    // The closest supported version before 8.5.0 was SortedSet fields, so we mimic that
+                    o.writeByte((byte) 0);
+                    o.writeOptionalBoolean(field.getMissingValue() == null ? null : field.getMissingValue() == SortField.STRING_FIRST);
+                    o.writeBoolean(true);
+                    o.writeBoolean(field.getReverse());
+                } else {
+                    o.writeByte((byte) 5);
+                    o.writeOptionalBoolean(field.getMissingValue() == null ? null : field.getMissingValue() == SortField.STRING_FIRST);
+                    o.writeBoolean(field.getReverse());
```

Comment:
Not sure if I understand this part very well. Don't we need to write a boolean for the selector? 

In all other cases (`type = 0`) I see one byte and thre booleans are written. Not sure why `type = 5` requires only 2 booleans.

Reply:
If we have a selector we use the other case above. In this case we just don't have one.